notebook/services/contents/filemanager.py

```diff
@@ -586,3 +590,22 @@ def get_kernel_path(self, path, model=None):
         else:
             parent_dir = ''
         return parent_dir
+
+    @staticmethod
+    def _validate_path(path):
+        """Checks if the path contains invalid characters relative to the current platform"""
+
+        if sys.platform == 'win32':
+            # On Windows systems, we MUST disallow colons otherwise an Alternative Data Stream will
+            # be created and confusion will reign! (See https://github.com/jupyter/notebook/issues/5190)
+            # Go ahead and add other invalid (and non-path-separator) characters here as well so there's
+            # consistent behavior - although all others will result in '[Errno 22]Invalid Argument' errors.
+            invalid_chars = (':', '>', '<', '*', '?', '|', '"')
+        else:
+            # On non-windows systems, allow the underlying file creation to perform enforcement when appropriate
+            invalid_chars = ()
+
+        for char in invalid_chars:
+            if char in path:
+                raise web.HTTPError(400, u"Path '{}' contains invalid characters {} relative to its platform.  "
```

Comment:
How about `Path '{}' contains characters that are invalid for the filesystem. Path names on this filesystem cannot contain any of the following characters: {}.`? I'm trying to make the message simpler, though it seems a little less actionable too (though in the validation method, it's not clear exactly who is calling us and what the recovery procedure should be...)

Reply:
Thank you - that reads much more cleanly.  Here's a screen shot with the update:
![Screen Shot 2020-01-28 at 3 25 12 PM](https://user-images.githubusercontent.com/22599560/73314363-864f9c00-41e2-11ea-9f35-3146323b31c1.png)

Previously, I was taking advantage of the parenthesis that sets print with.  That could now be argued as unnecessary.  If there's an easy pythonic way to remove those, I can do that - otherwise it seems okay to me.

Reply:
You could just define it as a string instead of a tuple.

Reply:
`invalid_chars = ':><*?|"'`